QSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1263 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "QSPIFBlockDevice.h"
+#include <string.h>
+#include "mbed_wait_api.h"
+
+/* Default QSPIF Parameters */
+/****************************/
+#define QSPIF_DEFAULT_READ_SIZE  1
+#define QSPIF_DEFAULT_PROG_SIZE  1
+#define QSPIF_DEFAULT_SE_SIZE    4096
+#define QSPI_MAX_STATUS_REGISTER_SIZE 2
+#define QSPI_STATUS_REGISTER_WRITE_TIMEOUT_MSEC 50
+#define QSPIF_DEFAULT_TIMEOUT_MSEC    1
```

Comment:
how to change these default values? Or shall they be changed? Can this msec be >1 in some cases? If yes, then this should be:

```
#ifndef QSPIF_DEFAULT_TIMEOUT_MSEC
#define QSPIF_DEFAULT_TIMEOUT_MSEC 1
#endif
```

A user can tweak it if needed

Reply:
Or maybe this should flash specific, of course only if operation time can be read from SFDP?

Reply:
@maciejbocianski and @0xc0170  : Thanks for this important remark. I looked at the link from Maciej and changed the logic - those  "DEFAULT" timeout macros are not really configurable and were removed. 

@maciejbocianski - regarding the erase/program typical/max time, although they are specified per erase type and program in SFDP, I think its a bit of an overkill to parse them and run the calculations for a fail timeout mechanism - instead I just entered a Max retries time of ~1 second that should cover the max for all failure cases. If we see this mechanism is too simple and doesn't cover most boards we can improve it in next phase.